Source/Particles/LaserParticleContainer.cpp

```diff
@@ -158,6 +159,9 @@ LaserParticleContainer::LaserParticleContainer (AmrCore* amr_core, int ispecies,
         m_position[0] += (Z0_boost-m_Z0_lab)*m_nvec[0];
         m_position[1] += (Z0_boost-m_Z0_lab)*m_nvec[1];
         m_position[2] += (Z0_boost-m_Z0_lab)*m_nvec[2];
+        AMREX_ALWAYS_ASSERT(m_v_antenna[0] == 0. && m_v_antenna[1] == 0.,
+                            "The antenna can only move in the z direction.");
+        m_v_antenna[2] = (m_v_antenna[2]-WarpX::beta_boost)/(1-WarpX::beta_boost*m_v_antenna[2]);
```

Comment:
Three things we need to take care here:
- This probably needs extra treatment for: 1D/2D/RZ :)
- Let's also use `.at()` here, so we have a runtime check for array accesses
- the `1-` needs a `_rt` so it's not taken as an integer literal, but a floating point literal
- 
```suggestion
        AMREX_ALWAYS_ASSERT(m_v_antenna.at(0) == 0. && m_v_antenna.at(1) == 0.,
                            "The antenna can only move in the z direction.");
        m_v_antenna.at(2) = (m_v_antenna.at(2)-WarpX::beta_boost)/(1_rt-WarpX::beta_boost*m_v_antenna.at(2));
```

Reply:
Thanks, I simplified this a little bit and incorporated your comments.